test/deploy/rackhd_stack_init.py

```diff
@@ -230,36 +230,33 @@ def test11_add_management_server(self):
         print "**** Creating management server."
         usr = ""
         pwd = ""
-        # find correct BMC passwords from global config
+        # find correct BMC passwords from credentials list
         for creds in fit_common.fitcreds()['bmc']:
-            if fit_common.remote_shell('ipmitool -I lanplus -H ' + fit_common.fitargs()['bmc']
+            if fit_common.remote_shell('ipmitool -I lanplus -H ' + fit_common.fitcfg()['bmc']
                                    + ' -U ' + creds['username'] + ' -P '
                                    + creds['password'] + ' fru')['exitcode'] == 0:
                 usr = creds['username']
                 pwd = creds['password']
         # create management node using these creds
-        payload = {
-                    "name": "Management Server",
-                    "type": "mgmt",
-                    "autoDiscover": "true",
-                    "ipmi-obm-service": {
-                        "host": fit_common.fitargs()['bmc'],
-                        "user": usr,
-                        "password": pwd
-                    }
-                    }
-        api_data = fit_common.rackhdapi("/api/2.0/nodes", action='post', payload=payload)
-        self.assertEqual(api_data['status'], 201, 'Incorrect HTTP return code, expecting 201, got '
-                         + str(api_data['status']))
-        # run discovery workflow
-        payload = {
-                    "name": "Graph.MgmtSKU.Discovery",
-                    "options":{"defaults": {"nodeId": api_data['json']['id']}}
-                    }
-        api_data = fit_common.rackhdapi("/api/2.0/nodes/" + api_data['json']['id'] + "/workflows",
-                                        action='post', payload=payload)
-        self.assertEqual(api_data['status'], 201, 'Incorrect HTTP return code, expecting 201, got '
-                         + str(api_data['status']))
+        if usr != "" and pwd != "":
+            payload = {
+                        "name":"Management Server " + str(time.time()),
+                        "type": "mgmt",
+                        "autoDiscover": "true",
+                        "obms": [
+                            { "service": "ipmi-obm-service",
+                            "config": { "host": fit_common.fitcfg()['bmc'],
+                                        "user": usr,
+                                        "password": pwd
+                                        }
+                                }
+                            ]
+                        }
+            api_data = fit_common.rackhdapi("/api/2.0/nodes", action='post', payload=payload)
+            self.assertEqual(api_data['status'], 201, 'Incorrect HTTP return code, expecting 201, got '
+                             + str(api_data['status']))
```

Comment:
This can be for later, but we really should have a tests here that the node actually did get created correctly.  The 201 status just tells us that the workflow got started.  Not that it completed successfully.